modules/datahandling/filerequests.go

```diff
@@ -86,10 +191,52 @@ func (f *fileMoveRequest) setAbstractRequest(req *abstractRequest) {
 	f.abstractRequest = *req
 }
 
-func (f fileMoveRequest) process() (response *serverMessageWrapper, notification *serverMessageWrapper, err error) {
-	// TODO
-	fmt.Printf("Recieved file move request from %s\n", f.SenderID)
-	return nil, nil, nil
+func (f fileMoveRequest) process(db dbfs.DBFS) ([]dhClosure, error) {
+	res := new(serverMessageWrapper)
+	res.Timestamp = time.Now().UnixNano()
+	res.Type = "Responce"
+
+	not := new(serverMessageWrapper)
+	not.Timestamp = res.Timestamp
+	not.Type = "Notification"
+
+	res.ServerMessage = response{
+		Status: fail,
+		Tag:    f.Tag,
+		Data:   struct{}{}}
+
+	fileMeta, err := db.MySQLFileGetInfo(f.FileID)
+	if err != nil {
+		return accumulate(toSenderClosure{msg: res}), err
+	}
+
+	not.RoutingKey = strconv.FormatInt(fileMeta.ProjectID, 10)
+	// TODO (normal/required): check if permission high enough on project (fileMeta.ProjectID)
+
+	err = db.MySQLFileMove(f.FileID, f.NewPath)
+	if err != nil {
+		return accumulate(toSenderClosure{msg: res}), err
+	}
+
+	err = db.FileMove(fileMeta.RelativePath, fileMeta.Filename, f.NewPath, fileMeta.Filename, fileMeta.ProjectID)
+	if err != nil {
+		return accumulate(toSenderClosure{msg: res}), err
+	}
+
+	res.ServerMessage = response{
+		Status: success,
+		Tag:    f.Tag,
+		Data:   struct{}{}}
```

Comment:
this is an anonymous type, so the `Data` field doesn't have a default value. also I'm not sure, but I don't think that the struct keyword itself has a default value because it's not really a type in and of itself, so `struct{}` doesnt work either.